src/main/java/com/ace/alfox/controllers/ActionController.java

```diff
@@ -3,32 +3,42 @@
 import com.ace.alfox.game.models.Player;
 import com.ace.alfox.lib.ActionFactory;
 import com.ace.alfox.lib.ActionResult;
-import com.ace.alfox.lib.PlayerFactory;
-import org.springframework.http.*;
+import com.ace.alfox.lib.Database;
+import org.dizitart.no2.NitriteId;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
 import org.springframework.web.bind.annotation.*;
 
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpSession;
-
-import java.io.*;
 import java.util.Map;
 
 @RestController
 class ActionController {
 
+    @Autowired
+    private Database db;
+    @Autowired
+    private ActionFactory actionFactory;
+
     @RequestMapping(value = "/perform/{command}", method = { RequestMethod.GET, RequestMethod.POST })
-    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) throws IOException {
+    public ResponseEntity home(HttpServletRequest request, @PathVariable String command, @RequestBody Map<String, Object> params) {
         HttpSession session = request.getSession(true);
-        String playerId = (String) session.getAttribute("cid");
-
-        Player player = PlayerFactory.fetch(playerId);
-        if(player == null) {
-            return ResponseEntity.notFound().build();
+        NitriteId playerId = (NitriteId) session.getAttribute("pid");
+
+        if(playerId == null) {
+            //return ResponseEntity.notFound().build();
+            // Instead of just returning 404, create a test player
+            Player _player = new Player();
+            db.players.insert(_player);
+            session.setAttribute("pid", _player.id);
+            playerId = _player.id;
         }
 
-        ActionFactory doSomething = ActionFactory.named(command).withParameters(params);
+        Player player = db.players.getById(playerId);
+        ActionFactory doSomething = actionFactory.named(command).withParameters(params);
         ActionResult result = player.applyAction(doSomething);
-        PlayerFactory.save(result.player);
+        db.players.update(player);
```

Comment:
In general I don't think we should expose a classes fields. For example, if we find that `db.players` is suddenly null, it's not easy to track down what code set it to null because anything with a reference to the instance of Database could have accidentally set it to null (or something else). Suddenly the Database class is no longer in control of its fields. Generally I find it's only ever good to expose fields that are immutable. If you hate boilerplate, there are ways around that. For example:

```java
class SomeDataClass {
    @Getter private final String someField;
}
```

This is a Lombok annotation that generates a Getter for this field. You could do the same with `@Setter`. Or you could annotate the whole class with `@Data` and you get getters/setters for all fields with only the one annotation. It will also generate the hashCode, toString, etc for you. I resisted Lombok hard at first, but I've found it makes code very easy to read and so much faster to write, and less buggy. Kind of like using Kotlin I suppose (I've never used Kotlin but I hear similar praises). I find it better than people generating hashCode() with their IDE because if you add a field you don't have to remember to re-generate, which is super easy to miss in a code review.

Anyway, don't feel like you have to change the params to private and start providing getters/setters for everything, it's not a big deal, but I wanted to bring it up. A good compromise if you're against the comments above would be to set these fields to final since I doubt you want things changing them.

Reply:
^This might be worth a good debate in person sometime! :)

Reply:
I agree that all fields exposed should be `final`, a mistake on my part. But I think more than anything I don't dislike the creation of getters, it's the usage of the getters.

A method call to me implies that processing is happening, when direct access to a field is simply navigational.

```java
// the difference between
db.players.getById(1234L);
// and
db.players().getById(1234L);
```

In the latter the getter implies fetching the entire result set, at least to me. I realize that this is just how it is in Java, I'm just used to other languages.

Reply:
Interesting! It's so common in Java that I never thought of that relationship. In general I always try to ensure my gets() are cheap, at least on subsequent calls. For example on the dataRepository.getWhatever() it could be an expensive network or database call, but if that's the case I'll cache it so that it's cheap. In this way consumers of my classes always know they are getting as performance gets() as possible. With a get() I can easily decide to make a param lazy, but if my interface is someone accessing the field then I need to populate in the constructor or have them check for initialization and then call some init() function.

I have found in Python and Java it seems super common to just make fields/properties accessible. I'm not even sure if there is a good way to hide them in Python, although maybe you can with underscores or something. I have vague memories of making things private, but maybe that was a dream haha.